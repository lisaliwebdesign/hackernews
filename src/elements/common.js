import React from "react";

const PageContainer = props => <div className="container">{props.content}</div>;

export { PageContainer };
