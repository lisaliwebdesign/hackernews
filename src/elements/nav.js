import React from "react";
import { Link } from "react-router-dom";

const NavBar = props => (
  <nav
    role="navigation"
    aria-label="main navigation"
    className="navbar navbar-guest"
  >
    <div className="container">
      <div className="navbar-brand">
        <a href="/" className="navbar-item">
          <h1 className="title has-text-centered is-2 hp-title">Hacker News</h1>
        </a>
        <button
          data-target="navMenu"
          className={
            props.toggle
              ? "button navbar-burger is-active"
              : "button navbar-burger"
          }
          onClick={() => props.toggleClick()}
        >
          <span /> <span /> <span />
        </button>
      </div>
      <div
        id="navMenu"
        className={props.toggle ? "navbar-menu is-active" : "navbar-menu"}
      >
        <div className="navbar-start">
          {props.content.map((item, index) => (
            <Link
              key={index}
              to={item.path}
              className={
                props.active === item.id
                  ? item.className + " active"
                  : item.className
              }
              id={item.id}
              onClick={() => props.onClick(item.id)}
            >
              {item.name}
            </Link>
          ))}
        </div>
        <div className="navbar-end">
          <div
            className={props.isLogInNav ? "navbar-item active" : "navbar-item"}
          >
            <Link to="/user/login" onClick={() => props.onClick("user-log-in")}>
              SIGN IN
            </Link>
          </div>
        </div>
      </div>
    </div>
  </nav>
);

export { NavBar };
