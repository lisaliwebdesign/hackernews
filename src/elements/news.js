import React, { Component, Fragment } from "react";
import { NewsItemTitle } from "../elements/news/title";
import { NewsItemPopUp } from "../elements/news/popup";
import { NewsItemHideIcon } from "../elements/news/hide";
import { NewsItemInfo } from "../elements/news/info";

const modal = item => {
  return <NewsItemPopUp item={item} />;
};

const NewsItemContainer = props => (
  <div className="sandbox">
    {props.content.map((item, index) => (
      <div
        key={index}
        id={item.id}
        className={"card news-item " + item.type + "-col"}
      >
        <article className="tile is-child notification is-white is-radiusless">
          <NewsItemTitle
            url={item.url !== undefined ? item.url : ""}
            title={item.title !== undefined ? item.title : ""}
          />
        </article>
        <NewsItemHideIcon />

        <div className="title-bottom">
          <NewsItemInfo
            score={item.score !== undefined ? item.score : ""}
            by={item.by !== undefined ? item.by : ""}
            time={item.time !== undefined ? item.time : new Date()}
          />
          <span className="item-right">
            {item.descendants
              ? modal(<i className="far fa-comment-dots" />)
              : modal(<i className="far fa-comment-dots icon-highlight" />)}
            {item.descendants}
          </span>
        </div>
      </div>
    ))}
  </div>
);

export { NewsItemContainer };
