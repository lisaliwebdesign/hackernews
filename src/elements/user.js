import React from "react";

const LogIn = props => (
  <div className="login-wrapper columns is-gapless">
    <div className="column is-7">
      <div className="hero is-fullheight">
        <div className="hero-heading">
          <div className="section has-text-centered">
            <h1 className="title has-text-centered is-2 hp-title">
              This page is under construction
            </h1>
          </div>

          <div className="no-account-link has-text-centered">
            <a href="#">Don't have an account ? </a>
          </div>
        </div>
        <div className="hero-body">
          <div className="container">
            <div className="columns">
              <div className="column" />
              <div className="column is-5">
                <form data-request="onSignin">
                  <div
                    id="signin-form"
                    className="login-form animated preFadeInLeft fadeInLeft"
                  >
                    <div className="field pb-10">
                      <div className="control">
                        <input
                          id="userSigninLogin"
                          className="input is-large"
                          type="text"
                          name="login"
                          placeholder="Enter your email"
                          required=""
                        />
                      </div>
                    </div>

                    <div className="field pb-20">
                      <div className="control">
                        <input
                          id="userSigninPassword"
                          className="input is-large"
                          type="password"
                          name="password"
                          placeholder="Enter your password"
                          required=""
                        />
                      </div>
                    </div>

                    <p className="control login">
                      <button
                        type="submit"
                        className="button button-cta primary-btn btn-align-lg btn-outlined is-bold is-fullwidth rounded raised no-lh  will-load"
                      >
                        Log in
                      </button>
                    </p>
                  </div>
                </form>

                <div id="partialUserResetForm">
                  <form
                    data-request="ninjaReset::onRestorePassword"
                    data-request-update="'ninjaReset::reset': '#partialUserResetForm'"
                  >
                    <div
                      id="recover-form"
                      className="login-form animated preFadeInLeft fadeInLeft is-hidden"
                    >
                      <h2 className="title is-4 has-text-centered">
                        Lost your Password ?
                      </h2>

                      <div className="field pb-20">
                        <div className="control">
                          <input
                            id="userRestoreEmail"
                            className="input is-large"
                            type="email"
                            name="email"
                            placeholder="Enter your email"
                            required=""
                          />
                        </div>
                      </div>

                      <p className="control login">
                        <button
                          type="submit"
                          className="button button-cta secondary-btn btn-align-lg btn-outlined is-bold is-fullwidth rounded raised no-lh will-load"
                        >
                          Restore password
                        </button>
                      </p>
                    </div>
                  </form>
                </div>

                <div
                  id="recover"
                  className="section forgot-password animated preFadeInLeft fadeInLeft"
                >
                  <p className="has-text-centered">
                    <a href="#">Forgot password ?</a>
                  </p>
                </div>

                <div
                  id="back-to-login"
                  className="section forgot-password animated preFadeInLeft fadeInLeft is-hidden"
                >
                  <p className="has-text-centered">
                    <a href="#">Back to Sign in</a>
                  </p>
                </div>
              </div>
              <div className="column" />
            </div>
          </div>
        </div>
      </div>
    </div>

    <div className="column login-column is-5 is-hidden-mobile hero-banner">
      <div className="hero is-fullheight is-theme-primary is-relative">
        <div className="columns has-text-centered">
          <div className="column" />
        </div>
        <img
          className="login-city"
          src="https://cssninja.io/themes/cssninja/assets/images/bg/temple.svg"
          alt=""
        />
      </div>
    </div>
  </div>
);
export { LogIn };
