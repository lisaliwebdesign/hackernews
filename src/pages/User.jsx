import React, { Fragment } from "react";
import Nav from "../components/Nav";
import { LogIn } from "../elements/user";

export default class User extends React.Component {
  render() {
    return (
      <Fragment>
        <Nav isLoginPage={true} /> <LogIn />
      </Fragment>
    );
  }
}
