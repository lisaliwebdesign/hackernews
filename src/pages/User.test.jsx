import User from "./User";
import { LogIn } from "../elements/user";
import { MemoryRouter as Router, withRouter } from "react-router-dom"; // 4.0.0

it("<Pages><User - renders component", () => {
  const Nav = { isLoginPage: true };
  const LogIn = mount(<div>Log in </div>);

  const wrapper = render(
    <Router>
      <User Nav={Nav} LogIn={LogIn} router={"test"} />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});
