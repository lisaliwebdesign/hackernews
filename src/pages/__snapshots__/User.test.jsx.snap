// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pages><User - renders component 1`] = `
Array [
  <nav
    aria-label="main navigation"
    class="navbar navbar-guest"
>
    <div
        class="container"
    >
        <div
            class="navbar-brand"
        >
            <a
                class="navbar-item"
                href="/"
            >
                <h1
                    class="title has-text-centered is-2 hp-title"
                >
                    Hacker News
                </h1>
            </a>
            <button
                class="button navbar-burger"
                data-target="navMenu"
            >
                <span>
                    
                </span>
                 
                <span>
                    
                </span>
                 
                <span>
                    
                </span>
            </button>
        </div>
        <div
            class="navbar-menu"
            id="navMenu"
        >
            <div
                class="navbar-start"
            >
                <a
                    class="navbar-item"
                    href="/news/topstories"
                    id="topstories"
                >
                    Top stories
                </a>
                <a
                    class="navbar-item"
                    href="/news/newstories"
                    id="newstories"
                >
                    New stories
                </a>
                <a
                    class="navbar-item"
                    href="/news/beststories"
                    id="beststories"
                >
                    Comments
                </a>
                <a
                    class="navbar-item"
                    href="/news/askstories"
                    id="askstories"
                >
                    Ask
                </a>
                <a
                    class="navbar-item"
                    href="/news/showstories"
                    id="showstories"
                >
                    Show
                </a>
                <a
                    class="navbar-item"
                    href="/news/jobstories"
                    id="jobstories"
                >
                    Job
                </a>
            </div>
            <div
                class="navbar-end"
            >
                <div
                    class="navbar-item active"
                >
                    <a
                        href="/user/login"
                    >
                        SIGN IN
                    </a>
                </div>
            </div>
        </div>
    </div>
</nav>,
  " ",
  <div
    class="login-wrapper columns is-gapless"
>
    <div
        class="column is-7"
    >
        <div
            class="hero is-fullheight"
        >
            <div
                class="hero-heading"
            >
                <div
                    class="section has-text-centered"
                >
                    <h1
                        class="title has-text-centered is-2 hp-title"
                    >
                        This page is under construction
                    </h1>
                </div>
                <div
                    class="no-account-link has-text-centered"
                >
                    <a
                        href="#"
                    >
                        Don't have an account ? 
                    </a>
                </div>
            </div>
            <div
                class="hero-body"
            >
                <div
                    class="container"
                >
                    <div
                        class="columns"
                    >
                        <div
                            class="column"
                        >
                            
                        </div>
                        <div
                            class="column is-5"
                        >
                            <form
                                data-request="onSignin"
                            >
                                <div
                                    class="login-form animated preFadeInLeft fadeInLeft"
                                    id="signin-form"
                                >
                                    <div
                                        class="field pb-10"
                                    >
                                        <div
                                            class="control"
                                        >
                                            <input
                                                class="input is-large"
                                                id="userSigninLogin"
                                                name="login"
                                                placeholder="Enter your email"
                                                type="text"
                                            >
                                                
                                            </input>
                                        </div>
                                    </div>
                                    <div
                                        class="field pb-20"
                                    >
                                        <div
                                            class="control"
                                        >
                                            <input
                                                class="input is-large"
                                                id="userSigninPassword"
                                                name="password"
                                                placeholder="Enter your password"
                                                type="password"
                                            >
                                                
                                            </input>
                                        </div>
                                    </div>
                                    <p
                                        class="control login"
                                    >
                                        <button
                                            class="button button-cta primary-btn btn-align-lg btn-outlined is-bold is-fullwidth rounded raised no-lh  will-load"
                                            type="submit"
                                        >
                                            Log in
                                        </button>
                                    </p>
                                </div>
                            </form>
                            <div
                                id="partialUserResetForm"
                            >
                                <form
                                    data-request="ninjaReset::onRestorePassword"
                                    data-request-update="'ninjaReset::reset': '#partialUserResetForm'"
                                >
                                    <div
                                        class="login-form animated preFadeInLeft fadeInLeft is-hidden"
                                        id="recover-form"
                                    >
                                        <h2
                                            class="title is-4 has-text-centered"
                                        >
                                            Lost your Password ?
                                        </h2>
                                        <div
                                            class="field pb-20"
                                        >
                                            <div
                                                class="control"
                                            >
                                                <input
                                                    class="input is-large"
                                                    id="userRestoreEmail"
                                                    name="email"
                                                    placeholder="Enter your email"
                                                    type="email"
                                                >
                                                    
                                                </input>
                                            </div>
                                        </div>
                                        <p
                                            class="control login"
                                        >
                                            <button
                                                class="button button-cta secondary-btn btn-align-lg btn-outlined is-bold is-fullwidth rounded raised no-lh will-load"
                                                type="submit"
                                            >
                                                Restore password
                                            </button>
                                        </p>
                                    </div>
                                </form>
                            </div>
                            <div
                                class="section forgot-password animated preFadeInLeft fadeInLeft"
                                id="recover"
                            >
                                <p
                                    class="has-text-centered"
                                >
                                    <a
                                        href="#"
                                    >
                                        Forgot password ?
                                    </a>
                                </p>
                            </div>
                            <div
                                class="section forgot-password animated preFadeInLeft fadeInLeft is-hidden"
                                id="back-to-login"
                            >
                                <p
                                    class="has-text-centered"
                                >
                                    <a
                                        href="#"
                                    >
                                        Back to Sign in
                                    </a>
                                </p>
                            </div>
                        </div>
                        <div
                            class="column"
                        >
                            
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <div
        class="column login-column is-5 is-hidden-mobile hero-banner"
    >
        <div
            class="hero is-fullheight is-theme-primary is-relative"
        >
            <div
                class="columns has-text-centered"
            >
                <div
                    class="column"
                >
                    
                </div>
            </div>
            <img
                alt=""
                class="login-city"
                src="https://cssninja.io/themes/cssninja/assets/images/bg/temple.svg"
            >
                
            </img>
        </div>
    </div>
</div>,
]
`;
